src/app/state/wallet/saga.ts

```diff
@@ -129,15 +127,17 @@ export function* openWalletFromMnemonic({ payload: mnemonic }: PayloadAction<str
  * If the wallet exists already, do nothingg
  * If it has "selectImmediately", we select it immediately
  */
-export function* addWallet({ payload: newWallet }: PayloadAction<AddWalletPayload>) {
+export function* addWallet({ payload }: PayloadAction<AddWalletPayload>) {
+  const { selectImmediately, ...newWallet } = payload
   const existingWallet = yield* call(getWalletByAddress, newWallet.address)
   if (!existingWallet) {
     yield* put(walletActions.walletOpened(newWallet))
   }
 
   const walletId = existingWallet ? existingWallet.id : newWallet.id
 
-  if (newWallet.selectImmediately) {
+  if (selectImmediately) {
+    yield* put(walletActions.selectWallet(undefined)) // Workaround so useRouteRedirects detects selecting the same account
```

Comment:
Why you have restored this line in a different place?

Reply:
To avoid bugs with importing multiple accounts